### DISCUSSION TITLE
Stabalization - roadmap to v1.0

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1392] [General]
Ah, massive github fail by me (@tcharding) I managed to edit this description then delete the revision while meaning to revert the edit. Please see https://github.com/rust-bitcoin/rust-secp256k1/issues/662

**Comment by Kixunil** [comment]
IMO the rerandomization issues need to be resolved because they may affect the API. Also we should fix some UB I found (well, it doesn't affect stability but it's not nice to have an otherwise polished lib with known UB). I vaguely remember seeing other things.

> Can we release v1.0 while secp256k1-sys remains on v0.x.y indefinitely

If and only if we do **not** re-export FFI types and functions which we currently do IIRC. I'd ACK hiding them or put them behind `unstable` feature gate.

> Should we wait till the context stuff is sorted out upstream before even talking about v1.0?

May help us but I don't think it's a strict blocker.

**Comment by apoelstra** [comment]
I do think the context stuff is a strict blocker. In my view to get 1.0 we need to:

* Audit any exports of FFI types (I think this is only in the `CPtr` trait which I'm happy to feature-gate, or even just document that it isn't stable and that normal users should never use it)
* Sort through the context/randomization discussions
* Fix the dependency/feature-gate mess (which means waiting for 1.62 I think to get `?` features)

In terms of total LOC diff I think we're really close. But unfortunately these are all big things that are at least partially outside of our control.

**Reply by Kixunil** [reply]
I just remembered maybe we should also resolve cooperation with the zkp fork. I had an idea to create `secp256k1-facade` crate that optionally depends on both of them (and doesn't claim to be one or the other) and depend on that from `bitcoin`.

**Comment by tcharding** [comment]
Make all API function use pass-by-value for `Copy` types (eg., `PublicKey`)

ref: https://github.com/rust-bitcoin/rust-bitcoin/issues/708
